bundles/org.openhab.binding.nanoleaf/src/main/java/org/openhab/binding/nanoleaf/internal/handler/NanoleafPanelHandler.java

```diff
@@ -166,33 +159,39 @@ private void stopAllJobs() {
 
     private void initializePanel(ThingStatusInfo panelStatus) {
         updateStatus(panelStatus.getStatus(), panelStatus.getStatusDetail());
+        updateState(CHANNEL_PANEL_COLOR, currentPanelColor);
         logger.debug("Panel {} status changed to {}-{}", this.getThing().getUID(), panelStatus.getStatus(),
                 panelStatus.getStatusDetail());
+
+        Bridge bridge = getBridge();
+        if (bridge != null) {
+            NanoleafControllerHandler handler = (NanoleafControllerHandler) bridge.getHandler();
+            if (handler != null) {
+                handler.getColorInformation().registerChangeListener(getPanelID(), this);
```

Comment:
To have both null and type safety, you could consider something like this:
```suggestion
            ThingHandler handler = bridge.getHandler();
            if (handler instanceof NanoleafControllerHandler) {
                ((NanoleafControllerHandler) handler).getColorInformation().registerChangeListener(getPanelID(), this);
```